timeseries/src/autogluon/timeseries/models/gluonts/callback.py

```diff
@@ -37,3 +38,73 @@ def on_epoch_end(
                 logger.warning("Time limit exceed during training, stop training.")
                 return False
         return True
+
+
+class GluonTSEarlyStoppingCallback(Callback):
+    """GluonTS callback to early stop the training if the validation loss
+    is not improved for `patience' round. For the GluonTS models used in autogluon,
+    the loss is always minimized.
+    """
+
+    def __init__(self, patience):
+        self.patience = patience
+        self.best_round = 0
+        self.best_loss = float("inf")
+
+    def on_validation_epoch_end(self, epoch_no, epoch_loss, **kwargs):
+        if epoch_loss < self.best_loss:
+            self.best_loss = epoch_loss
+            self.best_round = epoch_no
+            return True
+        else:
+            to_continue = (epoch_no - self.best_round) < self.patience
+            if not to_continue:
+                logger.warning(f"Early stopping triggered, stop training. Best epoch {self.best_round}")
+            return to_continue
+
+
+class GluonTSAdaptiveEarlyStoppingCallback(Callback):
+    """Adaptive early stopping where the patience is determined by `best_round' and `_update_patience' function.
+    Please also refer to autogluon.core.utils.early_stopping.py.
+
+    Parameters
+    ----------
+    adaptive_rate : float, default 0.3
+        The rate of increase in patience.
+        Set to 0 to disable, or negative to shrink patience during training.
+    adaptive_offset : int, default 10
+        The initial patience when cur_round is 0.
+    min_patience : int, default 10
+        The minimum value of patience.
+    max_patience : int, default 10000
+        The maximum value of patience.
+
+    Attributes
+    ----------
+    best_round : int
+        The most recent round passed to self.update with `is_best=True`.
+        Dictates patience and is used to determine if self.early_stop() returns True.
+    patience : int
+        If no improvement occurs in `patience` rounds or greater, self.early_stop will return True.
+        patience is dictated by the following formula:
+        patience = min(self.max_patience, (max(self.min_patience, round(self.best_round * self.adaptive_rate + self.adaptive_offset))))
+        Effectively, patience = self.best_round * self.adaptive_rate + self.adaptive_offset, bound by min_patience and max_patience
+    """
+
+    def __init__(self, adaptive_rate=0.3, adaptive_offset=10, min_patience=10, max_patience=10000):
+        self.es = AdaptiveES(adaptive_rate, adaptive_offset, min_patience, max_patience)
```

Comment:
is adaptive ES serializable? can we add a test for saving/loading models with AdaptiveES set?